Xamarin.Forms.Core/Shell/Shell.cs

```diff
@@ -1078,8 +1080,11 @@ internal void ProcessNavigated(ShellNavigatedEventArgs args)
 				{
 					baseShellItem.OnAppearing(() =>
 					{
-						OnNavigated(args);
-						Navigated?.Invoke(this, args);
+						Device.BeginInvokeOnMainThread(() =>
```

Comment:
@pictos What's the use case here?

Reply:
@PureWeen, for some reason, when the application initializes this event is not trigged (the breakpoint hits the `Navigated?.Invoke(this,args)` the listeners didn't fire), with that the CurrentPage will be null at the first time. After that, the event is trigged normally. So to fix this I force those calls to run on the UI thread.
If my explanation is confusing, please let me know.

Reply:
Boo! Alright, this is a great find but we'll want to figure out how to accomplish this same thing without using `Device.BeginInvokeOnMainThread`

Anything that requires `await Task.Delay(10)` or `Device.BeginInvokeOnMainThread` needs to be fixed so those aren't needed

Reply:
@PureWeen I agree, well I will dig in into the code and see if I can find something.